pkg/arvo/sys/arvo.hoon

```diff
@@ -156,6 +245,549 @@
+          [%slip vane=term card]
+        =/  vane
+          ~>  %mean.'bad-vane-label'
+          ?>  ((sane %tas) vane.q.vax)
+          vane.q.vax
+        ::
+        ::  yed: vase containing card
+        ::  hil: card as maze
+        ::
+        =^  xav  sac  (~(spot wa sac) 3 vax)
+        =^  yed  sac  (~(spot wa sac) 3 xav)
+        =^  hil  sac  (refine-card yed)
+        [[%slip vane hil] sac]
+      ::
+          [%hurl goof=^ ball=*]
+        =/  =goof
+          =/  mote  -.goof.q.vax
+          ?>  ?&  ?=(@ mote)
+                  ((sane %tas) mote)
+              ==
+          [mote ;;(tang +.goof.q.vax)]
+        ::
+        =^  bal  sac
+          =^  lab  sac  (~(spot wa sac) 7 vax)
+          $(vax lab)
+        ::
+        ?>  ?=(?(%pass %give) -.p.bal)
+        [[%hurl goof p.bal] sac]
+      ==
+    ::  +refine-card: card from vase
+    ::
+    ++  refine-card
+      |=  vax=vase
+      ^-  (pair maze worm)
+      ~>  %mean.'bad-card'
+      =^  hip  sac  (~(nell wa sac) p.vax)
+      ?>  hip
+      ?.  ?=(%meta -.q.vax)
+        ::
+        ::  for a non-meta card, the maze is the vase
+        ::
+        [[%& vax] sac]
+      ~>  %mean.'bad-meta'
+      ::
+      ::  tiv: vase of vase of card
+      ::  typ: vase of span
+      ::
+      =^  tiv  sac  (~(slot wa sac) 3 vax)
+      =^  hip  sac  (~(nell wa sac) p.tiv)
+      ?>  hip
+      =^  typ  sac  (~(slot wa sac) 2 tiv)
+      =.  sac  (~(neat wa sac) typ.vil [%& typ])
+      ::
+      ::  support for meta-meta-cards has been removed
+      ::
+      ?>  ?=(meta q.tiv)
+      [[%| q.tiv] sac]
+      ::
+      :: =/  mut
+      ::   ?>(?=(meta q.tiv) q.tiv)
+      :: |-  ^-  (pair [%| meta] worm)
+      :: ?.  ?=([%meta *] mut)
+      ::   [[%| mut] sac]
+      :: =^  dip  sac  (~(nets wa sac) mev.vil p.mut)
+      :: ?>  dip
+      :: $(q.tiv +.q.mut)
+    --
+  ::
+  ::  |va: vane engine
+  ::
+  ++  va
+    ~/  %va
+    |_  [our=ship vil=vile vax=vase sac=worm]
+    ::
+    ::  |plow:va: operate in time and space
+    ::
+    ++  plow
+      |=  [now=@da sky=slyd]
+      |%
+      ::  +peek:plow:va: read from a local namespace
+      ::
+      ++  peek
+        |=  [fur=(unit (set monk)) ren=@t bed=beam]
+        ^-  (unit (unit (cask meta)))
+        ::  namespace reads receive no entropy
+        ::
+        =/  sam=vane-sample  [our now *@uvJ sky]
+        =^  rig  sac
+          ~>  %mean.'peek: activation failed'
+          (~(slym wa sac) vax sam)
+        =^  gat  sac
+          ~>  %mean.'peek: call failed'
+          (~(slap wa sac) rig [%limb %scry])
+        ::
+        ;;  (unit (unit (cask meta)))
+        %+  slum  q.gat
+        ^-  scry-sample
+        :*  fur
+            ren
+            [%& p.bed]
+            q.bed
+            `coin`[%$ r.bed]
+            (flop s.bed)
+        ==
+      ::
+      ::  |spin:plow:va: move statefully
+      ::
+      ++  spin
+        |=  [hen=duct eny=@uvJ dud=(unit goof)]
+        :: =/  duc  !>(hen)
```

Comment:
I think this comment should either be removed or have context added.  I'm pretty sure I know what it means (that [duc.vil hen] is similar to !>(hen)), but I don't find it clarifying.

Reply:
Hedging against the `$vile` cached reflexives. I don't think they're actually necessary, but I need to work through any implications wrt type-of-type migrations first. I'll remove the comment (and replace `=*  err  !>(dud)` with a cons of a type from `$vile`).